### PR TITLE
Implementation of InfoBadge component

### PR DESCRIPTION
**Pull request by jbryknar**
## Pull request type

Please check the type of change your PR introduces:

- [x] Update
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes

## What is the current behavior?

Issue Number: #798

The current version of WpfUi does not have an InfoBadge implementation.

#### Added implementation of InfoBadge component according to Win UI 3.
https://learn.microsoft.com/en-us/windows/apps/design/controls/info-badge

#### The component can be used as a standalone component:
```xaml
<ui:InfoBadge Value="1" Severity="Attention" Opacity="1" Style="{DynamicResource ValueInfoBadgeStyle}>`
```
#### Supported PaneDisplayMode:

- Left (open and unopened panel, missing support for subitems in MenuItems yet)
- LeftMinimal (open and unopened panel)
- Fluent
- Top
- Bottom
- The component is implented into NavigationView, so it can be used in NavigationViewItem:
```xaml
<ui:NavigationViewItem Content="Items" Icon="{ui:SymbolIcon Library24}">
     <ui:NavigationViewItem.InfoBadge>
         <ui:InfoBadge Severity="Attention" Value="{Binding ViewModel.Value, Mode=OneWay}" Style="{DynamicResource ValueInfoBadgeStyle}" />
     </ui:NavigationViewItem.InfoBadge>
</ui:NavigationViewItem>
```
#### Added InfoBadge page to Status & info section for component demonstration

Snímek obrazovky 2023-12-09 203316

#### Basic parameters
- Value (Gets or sets the string to be displayed in an InfoBadge)
- Icon (Gets or sets the icon to be used in an InfoBadge). Primarily set for use with ui:SymbolIcon.
- Severity (Gets or sets the severity to be used in an InfoBadge)
- Opacity
- Style (Gets or sets the style of component, options are available as listed below)

#### Severity
Severity Name |Brush Name | Light | Dark
---|---|---|---
Attention | SystemFillColorAttention | #0078d4 | #4cc2ff
Informational | SystemFillColorInformational | #8a8a8a | #9d9d9d
Success | SystemFillColorSuccess | #0f7b0f | #6ccb5f
Caution | SystemFillColorCaution  | #9d5d00 | #fce100
Critical | SystemFillColorCritical | #c42b1c | #ff99a4

#### Styles
- IconInfoBadgeStyle (Icon is required parameter)
- ValueInfoBadgeStyle (Value is required parameter)
- DotInfoBadgeStyle

## Preview

Animace

The component can definitely be implemented better and more happily, so I'll be glad for the review and any comments. The goal was to help the author of the project and to contribute a bit to the development of the component.

### CODE REVIEW

File: src/Wpf.Ui/Resources/Theme/HC1.xaml [hunk 1]
```diff
@@ -350,7 +350,13 @@
     <SolidColorBrush x:Key="HyperlinkButtonForegroundDisabled" Color="{StaticResource SystemColorGrayTextColor}" />
 
     <!--  InfoBadge  -->
-    <!--  TODO  -->
+    <SolidColorBrush x:Key="InfoBadgeValueForeground" Color="{StaticResource TextFillColorInverse}" />
```
Comment (niels9001) [inline src/Wpf.Ui/Resources/Theme/HC1.xaml:353]: We can't be using `TextFillColorInverse` in HC themes. You'd problaby want to use `SystemColorButtonTextColor` instead

```suggestion
    <SolidColorBrush x:Key="InfoBadgeValueForeground" Color="{StaticResource SystemColorButtonTextColor}" />
```

File: src/Wpf.Ui/Resources/Theme/HC2.xaml [hunk 2]
```diff
@@ -349,7 +349,13 @@
     <SolidColorBrush x:Key="HyperlinkButtonForegroundDisabled" Color="{StaticResource SystemColorGrayTextColor}" />
 
     <!--  InfoBadge  -->
-    <!--  TODO  -->
+    <SolidColorBrush x:Key="InfoBadgeValueForeground" Color="{StaticResource TextFillColorInverse}" />
```
Comment (niels9001) [inline src/Wpf.Ui/Resources/Theme/HC2.xaml:352]: We can't be using `TextFillColorInverse` in HC themes. You'd problaby want to use `SystemColorButtonTextColor` instead

```suggestion
    <SolidColorBrush x:Key="InfoBadgeValueForeground" Color="{StaticResource SystemColorButtonTextColor}" />
```

File: src/Wpf.Ui/Resources/Theme/HCWhite.xaml [hunk 3]
```diff
@@ -349,7 +349,13 @@
     <SolidColorBrush x:Key="HyperlinkButtonForegroundDisabled" Color="{StaticResource SystemColorGrayTextColor}" />
 
     <!--  InfoBadge  -->
-    <!--  TODO  -->
+    <SolidColorBrush x:Key="InfoBadgeValueForeground" Color="{StaticResource TextFillColorInverse}" />
```
Comment (niels9001) [inline src/Wpf.Ui/Resources/Theme/HCWhite.xaml:352]: We can't be using `TextFillColorInverse` in HC themes. You'd problaby want to use `SystemColorButtonTextColor` instead

```suggestion
    <SolidColorBrush x:Key="InfoBadgeValueForeground" Color="{StaticResource SystemColorButtonTextColor}" />
```

File: src/Wpf.Ui/Resources/Theme/HCWhite.xaml [hunk 4]
```diff
@@ -349,7 +349,13 @@
...
+
+    <SolidColorBrush x:Key="InfoBadgeAttentionSeverityBackgroundBrush" Color="{StaticResource SystemFillColorAttention}" />
```
Comment (niels9001) [inline src/Wpf.Ui/Resources/Theme/HCWhite.xaml:354]:
```suggestion
    <SolidColorBrush x:Key="InfoBadgeAttentionSeverityBackgroundBrush" Color="{StaticResource SystemColorHighlightColor}" />
```

File: src/Wpf.Ui/Resources/Theme/HCWhite.xaml [hunk 5]
```diff
@@ -349,7 +349,13 @@
...
+    <SolidColorBrush x:Key="InfoBadgeInformationalSeverityBackgroundBrush" Color="{StaticResource SystemFillColorInformational}" />
```
Comment (niels9001) [inline src/Wpf.Ui/Resources/Theme/HCWhite.xaml:355]:
```suggestion
    <SolidColorBrush x:Key="InfoBadgeInformationalSeverityBackgroundBrush" Color="{StaticResource SystemColorHighlightColor}" />
```

File: src/Wpf.Ui/Resources/Theme/HCWhite.xaml [hunk 6]
```diff
@@ -349,7 +349,13 @@
...
+    <SolidColorBrush x:Key="InfoBadgeSuccessSeverityBackgroundBrush" Color="{StaticResource SystemFillColorSuccess}" />
+    <SolidColorBrush x:Key="InfoBadgeCautionSeverityBackgroundBrush" Color="{StaticResource SystemFillColorCaution}" />
```
Comment (niels9001) [inline src/Wpf.Ui/Resources/Theme/HCWhite.xaml:357]:
```suggestion
    <SolidColorBrush x:Key="InfoBadgeCautionSeverityBackgroundBrush" Color="{StaticResource SystemColorHighlightColor}" />
```

File: src/Wpf.Ui/Resources/Theme/HCWhite.xaml [hunk 7]
```diff
@@ -349,7 +349,13 @@
...
+    <SolidColorBrush x:Key="InfoBadgeCriticalSeverityBackgroundBrush" Color="{StaticResource SystemFillColorCritical}" />
```
Comment (niels9001) [inline src/Wpf.Ui/Resources/Theme/HCWhite.xaml:358]:
```suggestion
    <SolidColorBrush x:Key="InfoBadgeCriticalSeverityBackgroundBrush" Color="{StaticResource SystemColorHighlightColor}" />
```

File: src/Wpf.Ui/Resources/Theme/HCBlack.xaml [hunk 8]
```diff
@@ -349,7 +349,13 @@
     <SolidColorBrush x:Key="HyperlinkButtonForegroundDisabled" Color="{StaticResource SystemColorGrayTextColor}" />
 
     <!--  InfoBadge  -->
-    <!--  TODO  -->
+    <SolidColorBrush x:Key="InfoBadgeValueForeground" Color="{StaticResource TextFillColorInverse}" />
+
+    <SolidColorBrush x:Key="InfoBadgeAttentionSeverityBackgroundBrush" Color="{StaticResource SystemFillColorAttention}" />
```
Comment (niels9001) [inline src/Wpf.Ui/Resources/Theme/HCBlack.xaml:354]:
```suggestion
    <SolidColorBrush x:Key="InfoBadgeAttentionSeverityBackgroundBrush" Color="{StaticResource SystemColorHighlightColor}" />
```

File: src/Wpf.Ui/Resources/Theme/HCBlack.xaml [hunk 9]
```diff
@@ -349,7 +349,13 @@
...
+    <SolidColorBrush x:Key="InfoBadgeInformationalSeverityBackgroundBrush" Color="{StaticResource SystemFillColorInformational}" />
```
Comment (niels9001) [inline src/Wpf.Ui/Resources/Theme/HCBlack.xaml:355]:
```suggestion
    <SolidColorBrush x:Key="InfoBadgeInformationalSeverityBackgroundBrush" Color="{StaticResource SystemColorHighlightColor}" />
```

File: src/Wpf.Ui/Resources/Theme/HCBlack.xaml [hunk 10]
```diff
@@ -349,7 +349,13 @@
...
+    <SolidColorBrush x:Key="InfoBadgeSuccessSeverityBackgroundBrush" Color="{StaticResource SystemFillColorSuccess}" />
```
Comment (niels9001) [inline src/Wpf.Ui/Resources/Theme/HCBlack.xaml:356]:
```suggestion
    <SolidColorBrush x:Key="InfoBadgeSuccessSeverityBackgroundBrush" Color="{StaticResource SystemColorHighlightColor}" />
```

File: src/Wpf.Ui/Resources/Theme/HCBlack.xaml [hunk 11]
```diff
@@ -349,7 +349,13 @@
...
+    <SolidColorBrush x:Key="InfoBadgeCautionSeverityBackgroundBrush" Color="{StaticResource SystemFillColorCaution}" />
```
Comment (niels9001) [inline src/Wpf.Ui/Resources/Theme/HCBlack.xaml:357]:
```suggestion
    <SolidColorBrush x:Key="InfoBadgeCautionSeverityBackgroundBrush" Color="{StaticResource SystemColorHighlightColor}" />
```

File: src/Wpf.Ui/Resources/Theme/HCBlack.xaml [hunk 12]
```diff
@@ -349,7 +349,13 @@
...
+    <SolidColorBrush x:Key="InfoBadgeCriticalSeverityBackgroundBrush" Color="{StaticResource SystemFillColorCritical}" />
```
Comment (niels9001) [inline src/Wpf.Ui/Resources/Theme/HCBlack.xaml:358]:
```suggestion
    <SolidColorBrush x:Key="InfoBadgeCriticalSeverityBackgroundBrush" Color="{StaticResource SystemColorHighlightColor}" />
```

File: src/Wpf.Ui/Resources/Theme/HC2.xaml [hunk 13]
```diff
@@ -349,7 +349,13 @@
...
+
+    <SolidColorBrush x:Key="InfoBadgeAttentionSeverityBackgroundBrush" Color="{StaticResource SystemFillColorAttention}" />
```
Comment (niels9001) [inline src/Wpf.Ui/Resources/Theme/HC2.xaml:354]:
```suggestion
    <SolidColorBrush x:Key="InfoBadgeAttentionSeverityBackgroundBrush" Color="{StaticResource SystemColorHighlightColor}" />
```

File: src/Wpf.Ui/Resources/Theme/HC2.xaml [hunk 14]
```diff
@@ -349,7 +349,13 @@
...
+    <SolidColorBrush x:Key="InfoBadgeInformationalSeverityBackgroundBrush" Color="{StaticResource SystemFillColorInformational}" />
```
Comment (niels9001) [inline src/Wpf.Ui/Resources/Theme/HC2.xaml:355]:
```suggestion
    <SolidColorBrush x:Key="InfoBadgeInformationalSeverityBackgroundBrush" Color="{StaticResource SystemColorHighlightColor}" />
```

File: src/Wpf.Ui/Resources/Theme/HC2.xaml [hunk 15]
```diff
@@ -349,7 +349,13 @@
...
+    <SolidColorBrush x:Key="InfoBadgeSuccessSeverityBackgroundBrush" Color="{StaticResource SystemFillColorSuccess}" />
```
Comment (niels9001) [inline src/Wpf.Ui/Resources/Theme/HC2.xaml:356]:
```suggestion
    <SolidColorBrush x:Key="InfoBadgeSuccessSeverityBackgroundBrush" Color="{StaticResource SystemColorHighlightColor}" />
```

File: src/Wpf.Ui/Resources/Theme/HC2.xaml [hunk 16]
```diff
@@ -349,7 +349,13 @@
...
+    <SolidColorBrush x:Key="InfoBadgeCautionSeverityBackgroundBrush" Color="{StaticResource SystemFillColorCaution}" />
```
Comment (niels9001) [inline src/Wpf.Ui/Resources/Theme/HC2.xaml:357]:
```suggestion
    <SolidColorBrush x:Key="InfoBadgeCautionSeverityBackgroundBrush" Color="{StaticResource SystemColorHighlightColor}" />
```

File: src/Wpf.Ui/Resources/Theme/HC2.xaml [hunk 17]
```diff
@@ -349,7 +349,13 @@
...
+    <SolidColorBrush x:Key="InfoBadgeCriticalSeverityBackgroundBrush" Color="{StaticResource SystemFillColorCritical}" />
```
Comment (niels9001) [inline src/Wpf.Ui/Resources/Theme/HC2.xaml:358]:
```suggestion
    <SolidColorBrush x:Key="InfoBadgeCriticalSeverityBackgroundBrush" Color="{StaticResource SystemColorHighlightColor}" />
```

File: src/Wpf.Ui/Resources/Theme/HC1.xaml [hunk 18]
```diff
@@ -350,7 +350,13 @@
...
+
+    <SolidColorBrush x:Key="InfoBadgeAttentionSeverityBackgroundBrush" Color="{StaticResource SystemFillColorAttention}" />
```
Comment (niels9001) [inline src/Wpf.Ui/Resources/Theme/HC1.xaml:355]:
```suggestion
    <SolidColorBrush x:Key="InfoBadgeAttentionSeverityBackgroundBrush" Color="{StaticResource SystemColorHighlightColor}" />
```

File: src/Wpf.Ui/Resources/Theme/HC1.xaml [hunk 19]
```diff
@@ -350,7 +350,13 @@
...
+    <SolidColorBrush x:Key="InfoBadgeInformationalSeverityBackgroundBrush" Color="{StaticResource SystemFillColorInformational}" />
```
Comment (niels9001) [inline src/Wpf.Ui/Resources/Theme/HC1.xaml:356]:
```suggestion
    <SolidColorBrush x:Key="InfoBadgeInformationalSeverityBackgroundBrush" Color="{StaticResource SystemColorHighlightColor}" />
```

File: src/Wpf.Ui/Resources/Theme/HC1.xaml [hunk 20]
```diff
@@ -350,7 +350,13 @@
...
+    <SolidColorBrush x:Key="InfoBadgeSuccessSeverityBackgroundBrush" Color="{StaticResource SystemFillColorSuccess}" />
```
Comment (niels9001) [inline src/Wpf.Ui/Resources/Theme/HC1.xaml:357]:
```suggestion
    <SolidColorBrush x:Key="InfoBadgeSuccessSeverityBackgroundBrush" Color="{StaticResource SystemColorHighlightColor}" />
```

File: src/Wpf.Ui/Resources/Theme/HC1.xaml [hunk 21]
```diff
@@ -350,7 +350,13 @@
...
+    <SolidColorBrush x:Key="InfoBadgeCautionSeverityBackgroundBrush" Color="{StaticResource SystemFillColorCaution}" />
```
Comment (niels9001) [inline src/Wpf.Ui/Resources/Theme/HC1.xaml:358]:
```suggestion
    <SolidColorBrush x:Key="InfoBadgeCautionSeverityBackgroundBrush" Color="{StaticResource SystemColorHighlightColor}" />
```

File: src/Wpf.Ui/Resources/Theme/HC1.xaml [hunk 22]
```diff
@@ -350,7 +350,13 @@
...
+    <SolidColorBrush x:Key="InfoBadgeCriticalSeverityBackgroundBrush" Color="{StaticResource SystemFillColorCritical}" />
```
Comment (niels9001) [inline src/Wpf.Ui/Resources/Theme/HC1.xaml:359]:
```suggestion
    <SolidColorBrush x:Key="InfoBadgeCriticalSeverityBackgroundBrush" Color="{StaticResource SystemColorHighlightColor}" />
```

**Comment by jbryknar** [comment]
@niels9001 Thanks for the code review, theme files are not my cup of coffee. 😁

**Review by pomianowski — APPROVED** [review]

**Comment by pomianowski** [comment]
Hey @jbryknar, thanks for your work and contribution to the project, the `InfoBadge` looks great

**Pull request merged**